tools/run_tests/run_xds_tests.py

```diff
@@ -2238,6 +2239,135 @@ def _delay(pct):
+            return {
+                'priority': pri,
+                'matchRules': [{
+                    'prefixMatch':
+                        '/',
+                    'headerMatches': [{
+                        'headerName': testcase_header,
+                        'exactMatch': name,
+                    }],
+                }],
+                'service': original_backend_service.url,
+                'routeAction': {
+                    'retryPolicy': {
+                        'retryConditions': [retry_on],
+                        'numRetries': num_retries,
+                    },
+                },
+            }
+
+        route_rules = []
+        pri = 0
+        for (testcase_name, test_config,
+             expected_succeeded_rpcs_range) in test_cases:
+            (retry_policy, rpc_behavior) = test_config
+            route_rules.append(_route(pri, testcase_name, retry_policy))
+            pri += 1
+
+        set_validate_for_proxyless(gcp, False)
+        patch_url_map_backend_service(gcp,
+                                      original_backend_service,
+                                      route_rules=route_rules)
+
+        first_case = True
+        for (testcase_name, test_config,
+             expected_succeeded_rpcs_range) in test_cases:
+            (retry_policy, rpc_behavior) = test_config
+            logger.info('starting case %s', testcase_name)
+
+            client_config = {}
+            client_config['metadata'] = [
+                (messages_pb2.ClientConfigureRequest.RpcType.UNARY_CALL,
+                 testcase_header, testcase_name),
+                (messages_pb2.ClientConfigureRequest.RpcType.UNARY_CALL,
+                 'rpc-behavior', rpc_behavior),
+            ]
+            client_config['rpc_types'] = [
+                messages_pb2.ClientConfigureRequest.RpcType.UNARY_CALL,
+            ]
+            configure_client(**client_config)
+            # wait a second to help ensure the client stops sending RPCs with
+            # the old config.  We will make multiple attempts if it is failing,
+            # but this improves confidence that the test is valid if the
+            # previous client_config would lead to the same results.
+            time.sleep(1)
+            # Each attempt takes 10 seconds; 20 attempts is equivalent to 200
+            # second timeout.
+            attempt_count = 20
+            if first_case:
+                attempt_count = 120
+                first_case = False
+            for i in range(attempt_count):
+                logger.info('%s: attempt %d', testcase_name, i)
+
+                # wait for 5 more seconds to allow retry complete
+                stats = get_client_stats(num_rpcs, test_runtime_secs + 5)
+                succeeded_rpcs = sum(stats.rpcs_by_peer.values())
+                success = True
+                (expected_range_from,
+                 expected_range_to) = expected_succeeded_rpcs_range
```

Comment:
From the test case setup, it doesn't seem necessary to have a range, if "from" is always equal to "to".

Reply:
Right, the range was used for retry circuit breaker test case, now that the test case is removed, I can stop using range.